Sources/ProjectSpec/SpecFile.swift

```
@@ -64,8 +68,16 @@ public struct SpecFile {
         let jsonDictionary = try SpecFile.loadDictionary(path: path)
 
         let includes = Include.parse(json: jsonDictionary["include"])
-        let subSpecs: [SpecFile] = try includes.map { include in
-            try SpecFile(include: include, basePath: basePath, relativePath: relativePath)
+        let subSpecs: [SpecFile] = try includes.compactMap { include in
+            do {
+                return try SpecFile(include: include, basePath: basePath, relativePath: relativePath)
+            } catch {
+                if include.optional {
+                    return nil
```

This would make an optional includes silently fail if there was an different error than simply a missing file, for example if one of the includes own includes was missing. Is that the intention? Maybe it's better to only return nil if the file is actually missing

---

> Maybe it's better to only return nil if the file is actually missing

Do you mean `optional` should ignore throwing from `try path.read()` ?
https://github.com/yonaskolb/XcodeGen/blob/15a4f6abe601cbe5750276f93b572bd723e45912/Sources/ProjectSpec/SpecFile.swift#L89